### PR TITLE
handle empty RepoTags for ListImage

### LINKED ISSUE #155: panic when list image that be pulled by digest.
panic: runtime error: index out of range

I can fix it.

Log is below.
```
root@ubuntu:/home/cloud/go/src/github.com/kubernetes-incubator/cri-containerd# crictl pull docker.io/library/busybox@sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642
DEBU[0000] PullImageRequest: &PullImageRequest{Image:&ImageSpec{Image:docker.io/library/busybox@sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642,},Auth:nil,SandboxConfig:nil,} 
DEBU[0002] PullImageResponse: &PullImageResponse{ImageRef:sha256:c30178c5239f2937c21c261b0365efcda25be4921ccb95acd63beeeb78786f27,} 
Image is update to date for sha256:c30178c5239f2937c21c261b0365efcda25be4921ccb95acd63beeeb78786f27
root@ubuntu:/home/cloud/go/src/github.com/kubernetes-incubator/cri-containerd# crictl images
DEBU[0000] ListImagesRequest: &ListImagesRequest{Filter:&ImageFilter{Image:&ImageSpec{Image:,},},} 
DEBU[0000] ListImagesResponse: &ListImagesResponse{Images:[&Image{Id:sha256:c30178c5239f2937c21c261b0365efcda25be4921ccb95acd63beeeb78786f27,RepoTags:[],RepoDigests:[docker.io/library/busybox@sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642],Size_:701277,Uid:nil,Username:,}],} 
panic: runtime error: index out of range

goroutine 1 [running]:
main.glob..func10(0xc4200e4780, 0x0, 0xc4200e4780)
	/home/cloud/go/src/github.com/kubernetes-incubator/cri-tools/cmd/crictl/image.go:135 +0x5b1
github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli.HandleAction(0x1375cc0, 0x15c3870, 0xc4200e4780, 0xc42006f400, 0x0)
	/home/cloud/go/src/github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli/app.go:485 +0xd4
github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli.Command.Run(0x155c234, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x156176e, 0xb, 0x0, ...)
	/home/cloud/go/src/github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli/command.go:207 +0xb72
github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli.(*App).Run(0xc420087ba0, 0xc42000c1e0, 0x2, 0x2, 0x0, 0x0)
	/home/cloud/go/src/github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli/app.go:250 +0x7d0
main.main()
	/home/cloud/go/src/github.com/kubernetes-incubator/cri-tools/cmd/crictl/main.go:195 +0xaaf
```

### PR DESCRIPTION
**Pull request by yanxuean**
fix #155
Signed-off-by: yanxuean <yan.xuean@zte.com.cn>

### CODE REVIEW

File: cmd/crictl/image.go [hunk 1]
```diff
@@ -125,11 +125,7 @@ var listImageCommand = cli.Command{
 					printHeader = false
 					fmt.Fprintln(w, "IMAGE\tTAG\tIMAGE ID\tSIZE")
 				}
-				repoTags := "<none>"
-				if image.RepoTags != nil {
```
Comment (Random-Liu) [inline cmd/crictl/image.go]: We should change this to `if len(image.RepoTags) > 0`.

File: cmd/crictl/image.go [hunk 2]
```diff
@@ -270,6 +266,21 @@ func getAuth(creds string) (*pb.AuthConfig, error) {
 	}, nil
 }
 
+// Support these cases as below:
```
Comment (Random-Liu) [inline cmd/crictl/image.go]: I think it's fine to assume that this should always be `image:tag` or `image:digest`.

File: cmd/crictl/image.go (hunk 1 above)
Comment (yanxuean) [reply cmd/crictl/image.go]: Only this is insufficient. repoTagsPair[1] will still cause panic  when it is "<none>" or "tag1"

File: cmd/crictl/image.go (hunk 2 above)
Comment (yanxuean) [reply cmd/crictl/image.go]: Sorry, what do you mean?

Comment (Random-Liu) [reply cmd/crictl/image.go]: I think if the repotag is not `image:tag`, we could log an error. Ideally repo tag should always be `image:tag`.

Comment (yanxuean) [reply cmd/crictl/image.go]: Do you mean the cri-containerd or docker-shim should return "image:tag"?
In any case crictl should not panic. Crictl need to protect self.

**Comment by yanxuean** [comment]
Who can tell me How trigger a new test?  I need to do it when the reason of fail checking is belong to me.

File: cmd/crictl/image.go [hunk 3]
```diff
@@ -125,11 +125,7 @@ var listImageCommand = cli.Command{
 					printHeader = false
 					fmt.Fprintln(w, "IMAGE\tTAG\tIMAGE ID\tSIZE")
 				}
-				repoTags := "<none>"
-				if image.RepoTags != nil {
-					repoTags = image.RepoTags[0]
-				}
-				repoTagsPair := strings.Split(repoTags, ":")
+				repoTagsPair := parseRepoTagPair(image.RepoTags)
```
Comment (Random-Liu) [inline cmd/crictl/image.go]: normalizeRepoTagPair.

Comment (yanxuean) [reply cmd/crictl/image.go]: The name is perfect.

**Comment by yanxuean** [comment]
docker image for image pulled by repoDigest
```
root@ubuntu:/home/cloud/go/src/k8s.io/kubernetes# docker pull gcr.io/google_containers/pause@sha256:0d093c962a6c2dd8bb8727b661e2b5f13e9df884af9945b4cc7088d9350cd3ee
sha256:0d093c962a6c2dd8bb8727b661e2b5f13e9df884af9945b4cc7088d9350cd3ee: Pulling from google_containers/pause
a3ed95caeb02: Pull complete 
f11233434377: Pull complete 
Digest: sha256:0d093c962a6c2dd8bb8727b661e2b5f13e9df884af9945b4cc7088d9350cd3ee
Status: Downloaded newer image for gcr.io/google_containers/pause@sha256:0d093c962a6c2dd8bb8727b661e2b5f13e9df884af9945b4cc7088d9350cd3ee
root@ubuntu:/home/cloud/go/src/k8s.io/kubernetes# docker images
REPOSITORY                       TAG                 IMAGE ID            CREATED             SIZE
jpetazzo/nsenter                 latest              c16fe938c1a5        15 months ago       371MB
gcr.io/google_containers/pause   <none>              99e59f495ffa        17 months ago       747kB
root@ubuntu:/home/cloud/go/src/k8s.io/kubernetes# 
```

File: cmd/crictl/image.go (hunk 3 above)
Comment (yanxuean) [reply cmd/crictl/image.go]: done

File: cmd/crictl/image.go (hunk 2 above)
Comment (feiskyer) [reply cmd/crictl/image.go]: @yanxuean In which case repoTags will be `:tag2` or `tag1:`?

Comment (yanxuean) [reply cmd/crictl/image.go]: I don't know :)
I only protect self.

**Comment by Random-Liu** [comment]
For image which only has repo digest, we should:
1) Add `--digests` option to show image digest.
2) Show name with `<none>` tag if `--digests` is not specified.

I'm fine with adding `--digests` in another PR, but in this PR can we include the image name if a image has repo digest but not repo tag?

**Comment by feiskyer** [comment]
> include the image name if a image has repo digest but not repo tag

+1. @yanxuean Could you add this in this PR?

**Comment by yanxuean** [comment]
I am adding it. :)

**Comment by yanxuean** [comment]
@Random-Liu @feiskyer   PTAL
I don't know how to test the case for one image with multi repoTag.

**Comment by feiskyer** [comment]
LGTM, thanks

image

**Comment by feiskyer** [comment]
Also find a bug in dockershim, see https://github.com/kubernetes/kubernetes/issues/54122

**Comment by yanxuean** [comment]
@feiskyer  Tks

**Pull request merged**